l10n_ch_payment_slip/models/invoice.py

```diff
@@ -169,3 +171,24 @@ def _check_bvr_generatable(self):
                 errors.append(invoice + '\n'.join(msg))
         if errors:
             raise exceptions.UserError('\n'.join(errors))
+
+    @api.multi
+    def action_invoice_draft(self):
+        res = super().action_invoice_draft()
+        # Delete former printed payment slip
+        ActionReport = self.env['ir.actions.report']
+        try:
+            report_payment_slip = ActionReport._get_report_from_name(
+                'l10n_ch_payment_slip.one_slip_per_page_from_invoice')
+        except IndexError:
+            report_payment_slip = False
+        if report_payment_slip and report_payment_slip.attachment:
+            for invoice in self:
+                with invoice.env.do_in_draft():
```

Comment:
Q: why in draft mode? You are not changing anything, no?

Reply:
it comes from https://github.com/OCA/l10n-switzerland/pull/411 I don't want improve it in this PR

Reply:
perfectly fine :+1: but please leave comments in all the places where it would be nice to rethink/refactor in the future, thanks!